src/test/compile-fail/borrowck/borrowck-borrow-overloaded-auto-deref-mut.rs

```diff
@@ -99,7 +99,7 @@ fn assign_field1<'a>(x: Own<Point>) {
 }
 
 fn assign_field2<'a>(x: &'a Own<Point>) {
-    x.y = 3; //~ ERROR cannot assign
+    x.y = 3; //~ ERROR cannot borrow
```

Comment:
Both cases are fine. The special casing that caused `Deref::deref` to be chosen is just plain unnecessary.